libraries/standard/retry-utils/source/include/retry_utils.h

```diff
@@ -0,0 +1,174 @@
+ */
+typedef enum RetryUtilsStatus
+{
+    RetryUtilsSuccess = 0,     /**< @brief The function successfully calculated the next back-off value. */
+    RetryUtilsRngFailure = 1,  /**< @brief The function encountered failure in generating random number. */
+    RetryUtilsRetriesExhausted /**< @brief The function exhausted all retry attempts. */
+} RetryUtilsStatus_t;
+
+/**
+ * @brief Represents parameters required for calculating the back-off delay for the
+ * next retry attempt.
+ */
+typedef struct RetryUtilsContext
+{
+    /**
+     * @brief The maximum backoff delay (in milliseconds) between consecutive retry attempts.
+     */
+    uint16_t maxBackOffDelay;
+
+    /**
+     * @brief The total number of retry attempts completed.
+     * This value is incremented on every call to #RetryUtils_GetNextBackOff API.
+     */
+    uint32_t attemptsDone;
+
+    /**
+     * @brief The maximum backoff value (in milliseconds) for the next retry attempt.
+     */
+    uint16_t nextJitterMax;
+
+    /**
+     * @brief The maximum number of retry attempts.
+     */
+    uint32_t maxRetryAttempts;
+
+    /**
+     * @brief The random number generator function used for calculating the
+     * backoff value for the next retry attempt.
+     */
+    RetryUtils_RNG_t pRng;
+} RetryUtilsContext_t;
+
+/**
+ * @brief Initializes the context for using retry utils. The parameters
+ * are required for calculating the next retry backoff delay.
+ * This function must be called by the application before the first new retry attempt.
+ *
+ * @param[out] pRetryParams The context to initialize with parameters required
+ * for the next backoff delay calculation function.
+ * @param[in] maxBackOff The maximum backoff delay (in milliseconds) between
+ * consecutive retry attempts.
+ * @param[in] backOffBase The base value (in milliseconds) of backoff delay to
+ * use in the exponential backoff and jitter model.
+ * @param[in] maxAttempts The maximum number of retry attempts. Set the value to
+ * #RETRY_FOREVER to retry for ever.
+ * @param[in] pRng The platform-specific function to use for random number generation.
+ * The random number generator should be seeded before calling this function.
+ */
+/* @[define_retryutils_initializeparams] */
+void RetryUtils_InitializeParams( RetryUtilsContext_t * pContext,
+                                  uint16_t backOffBase,
+                                  uint16_t maxBackOff,
+                                  uint32_t maxAttempts,
+                                  RetryUtils_RNG_t pRng );
+/* @[define_retryutils_initializeparams] */
+
+/**
+ * @brief Simple exponential backoff and jitter function that provides the
+ * delay value for the next retry attempt.
+ * After a failure of an operation that needs to be retried, the application
+ * should use this function to obtain the backoff delay value for the next retry,
+ * and then wait for the backoff value before retrying the operation.
```

Comment:
Will it be better to mention explicitly to `sleep` instead of `wait` here?

Reply:
I didn't mention `sleep` to avoid being POSIX-specific language here.. I think `wait` is more generic than `sleep`. The `sleep` operation is one way to implement a timed wait

Reply:
ok. Sounds good to me.